codegen/src/types/derives.rs

```diff
@@ -83,18 +92,24 @@ impl Derives {
     pub fn insert(&mut self, derive: syn::Path) {
         self.derives.insert(derive);
     }
-}
 
-impl Default for Derives {
-    fn default() -> Self {
+    /// Creates a default instance of `Derives` with the `crate_path` prepended
+    /// to the set of default derives that reside in `subxt`.
+    pub fn default_with_crate_path(crate_path: &CratePath) -> Self {
         let mut derives = HashSet::new();
-        derives.insert(syn::parse_quote!(::subxt::ext::codec::Encode));
-        derives.insert(syn::parse_quote!(::subxt::ext::codec::Decode));
+        derives.insert(syn::parse_quote!(#crate_path::ext::codec::Encode));
+        derives.insert(syn::parse_quote!(#crate_path::ext::codec::Decode));
         derives.insert(syn::parse_quote!(Debug));
         Self { derives }
     }
 }
 
+impl Default for Derives {
+    fn default() -> Self {
+        Derives::default_with_crate_path(&CratePath::default())
+    }
+}
```

Comment:
Should this exist? I think we should make sure crate path is always explicitly provided so we are forced to pass it through always.

Reply:
So to be on the same page: would this go through until the user-facing side? I.e. would each user always be required to specify `::subxt` as the crate path to use?